micromamba/src/shell.cpp

```diff
@@ -33,19 +32,78 @@ namespace
             ->check(CLI::IsMember(std::set<std::string>(
                 { "bash", "posix", "powershell", "cmd.exe", "xonsh", "zsh", "fish", "tcsh", "dash" }
             )));
+    }
+
+    void init_root_prefix_option(CLI::App* subcmd, Configuration& config)
+    {
+        auto& root = config.at("root_prefix");
+        subcmd->add_option(
+            "root_prefix,-r,--root-prefix",
+            root.get_cli_config<fs::u8path>(),
+            root.description()
+        );
+    }
+
+    void init_prefix_options(CLI::App* subcmd, Configuration& config)
+    {
+        auto& prefix = config.at("target_prefix");
+        auto* prefix_cli = subcmd->add_option(
+            "-p,--prefix",
+            prefix.get_cli_config<fs::u8path>(),
+            prefix.description()
+        );
 
-        auto& prefix = config.insert(
-            Configurable("shell_prefix", std::string(""))
+        auto& name = config.at("env_name");
+        auto* name_cli = subcmd
+                             ->add_option(
+                                 "-n,--name",
+                                 name.get_cli_config<std::string>(),
+                                 name.description()
+                             )
+                             ->excludes(prefix_cli);
+
+        auto& prefix_or_name = config.insert(
+            Configurable("prefix_or_name", std::string(""))
                 .group("cli")
-                .description("The root prefix to configure (for init and hook), and the prefix "
-                             "to activate for activate, either by name or by path"),
+                .description("The prefix to activate, either by name or by path"),
             true
         );
-        subcmd->add_option(
-            "prefix,-p,--prefix,-n,--name",
-            prefix.get_cli_config<std::string>(),
-            prefix.description()
+
+        subcmd
+            ->add_option(
+                "prefix_or_name",
```

Comment:
Hmm, do we want to allow this? Is it for specific cases, or to follow e.g conda's behavior, or something else?

Reply:
Yes, at least for backward compatibility. It is a feature that you can do `micromamba activate env` without specifying.